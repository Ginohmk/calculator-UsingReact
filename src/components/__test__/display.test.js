/* eslint-disable comma-dangle */
import React from 'react';
import { screen, render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Display from '../display';

afterEach(cleanup);
describe('Test Display Component', () => {
  test('Test Components  Rendering for Display', () => {
    render(<Display curOperand="78" prevOperand="5" operation="+" />);

    const displayElement = screen.getByTestId('display-data');
    const displayContent = displayElement.textContent;

    expect(displayContent).toBe('78');
  });

  test('Testing Diplay snapShot ', () => {
    const tree = renderer
      .create(<Display prevOperand="3" operation="+" curOperand="78" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
