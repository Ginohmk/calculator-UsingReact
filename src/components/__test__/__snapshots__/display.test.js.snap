// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Display Component Testing Diplay snapShot  1`] = `
<div
  className="output"
>
  <p
    className="top-display"
  >
    3
    +
  </p>
  <p
    className="down-display "
    data-testid="display-data"
  >
    78
  </p>
</div>
`;
